### PR TITLE
Fix Max Move effects not activating against substitute

### PR DESCRIPTION
**Pull request by 420Blazeitt**

### CODE REVIEW

File: data/moves.js [hunk 1]
```diff
@@ -7143,7 +7147,9 @@ let BattleMovedex = {
 		priority: 0,
 		flags: {},
 		isMax: "Orbeetle",
-		pseudoWeather: 'gravity',
+		self: {
+			pseudoWeather: 'gravity',
```
Comment (Zarel) [inline data/moves.js]: In theory, this one shouldn't be necessary (nor the others).

Comment (420Blazeitt) [reply data/moves.js]: it is necessary, I tested the current code on PS and here are the results: https://replay.pokemonshowdown.com/gen8customgame-1012218343

File: data/moves.js [hunk 2]
```diff
@@ -7074,10 +7076,12 @@ let BattleMovedex = {
...
 		flags: {},
 		isMax: "Alcremie",
-		onAfterHit(target, source) {
-			for (let pokemon of source.side.active) {
-				pokemon.heal(pokemon.maxhp);
-			}
+		self: {
+			onAfterHit(target, source) {
+				for (let pokemon of source.side.active) {
+					pokemon.heal(pokemon.maxhp);
```
Comment (urkerab) [inline data/moves.js]: While you're here, this should be `this.heal(pokemon.maxhp, pokemon, pokemon);`. See https://www.smogon.com/forums/posts/8287090/

Comment (420Blazeitt) [reply data/moves.js]: thanks!

Comment (420Blazeitt) [reply data/moves.js]: thanks!

File: data/moves.js (hunk 1 above)
Comment (Zarel) [reply data/moves.js]: Awkward. I guess this will need to be fixed later.

Comment (Zarel) [reply data/moves.js]: Nice creative hack, though.

**Pull request merged**